### PR TITLE
refactor: update commander + remove deprecated commands

### PR DESCRIPTION
**Pull request by thekip**
# Description

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to Lingui?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

Fixes # (issue)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/cli/src/api/utils.ts [hunk 1]
```diff
@@ -11,43 +9,6 @@ export function prettyOrigin(origins: [filename: string, line?: number][]) {
...
- *    :param: command - command passed to CLI
- *    :param: availableCommands - all commands defined in commander.js
- *
- *    If unknown commands is passed to CLI, check it agains all available commands
- *    for possible misspelled letter. Output help with suggestions to console.
- */
-export function helpMisspelledCommand(command: string, availableCommands = []) {
-  const commandNames = availableCommands.map((command) => command.name())
-
-  // if no command is supplied, then commander.js shows help automatically
```
Comment (thekip) [inline packages/cli/src/api/utils.ts:25]: fresh commander do it out of the box

File: packages/cli/src/lingui-compile.ts [hunk 2]
```diff
@@ -180,41 +192,35 @@ if (require.main === module) {
     })
     .parse(process.argv)
 
-  const config = getConfig({ configPath: program.config })
+  const options = program.opts<CliOptions>()
```
Comment (thekip) [inline packages/cli/src/lingui-compile.ts:193]: this is biggest difference in versions, 

https://github.com/tj/commander.js/blob/master/CHANGELOG.md#migration-tips-1

File: packages/cli/src/lingui-extract.ts [hunk 3]
```diff
@@ -23,12 +23,6 @@ export default async function command(
   config: LinguiConfigNormalized,
   options: Partial<CliExtractOptions>
 ): Promise<boolean> {
-  // `react-app` babel plugin used by CRA requires either BABEL_ENV or NODE_ENV to be
-  // set. We're setting it here, because lingui macros are going to use them as well.
-  if (!process.env.BABEL_ENV && !process.env.NODE_ENV) {
-    process.env.BABEL_ENV = "development"
-  }
-
```
Comment (thekip) [inline packages/cli/src/lingui-extract.ts:31]: this is not needed

Comment (Martin005) [reply packages/cli/src/lingui-extract.ts:31]: Shouldn't this be removed also from the `lingui-extract-template.ts`?

Comment (thekip) [reply packages/cli/src/lingui-extract.ts:31]: done

**Review by Martin005 — APPROVED** [review]

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**